Set up continuous integration with github actions

leomartinez2020 opened the pull request:
This PR implements continuous integration with github actions

holgiosalos commented on .github/main.yml:
```diff
@@ -0,0 +1,22 @@
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+      - name: Install cypress
+        run: npm install cypress@latest
+      - name: Install linter
+        run: npm install eslint --save-dev
+      - name: Install linter cypress
+        run: npm install eslint-plugin-cypress --save-dev
```
these installs are not needed, as the dependencies are already specified in package.json file, the only required step is to run the ci:
```
      - name: Install
        run: npm ci
```

holgiosalos commented on .github/main.yml:
```diff
@@ -0,0 +1,22 @@
+name: GitHub Actions Demo
+on: [push]
```
currently there are no jobs or checks executed for this PR, review if this `on` trigger needs to be changed by a:
`on: [pull_request]`. 

Also review that you have enabled the Require status check option in the branch protection rules:
![image](https://user-images.githubusercontent.com/10209180/180567494-2996fe5e-ead3-405d-b9d6-9ea18f9b9cfe.png)

leomartinez2020 commented:
@holgiosalos I added the necessary changes to `main.yml` and updated the branch rules as seen in the image:
![cypress-ci](https://user-images.githubusercontent.com/5594381/180576576-b40feb34-11e1-4a1d-af53-3a6975c26636.png)

holgiosalos reviewed (commented):
not sure, but, something is not well configured yet, since the checks tab, does not have active jobs running:
![image](https://user-images.githubusercontent.com/10209180/180577973-bf82df79-b7f4-4d12-a9a6-50948f9027bc.png)

leomartinez2020 commented:
@holgiosalos <p><s>in that case I will have to create a new PR so that the check is triggered on pull_request as expected</s></p>

Update: I found the mistake: `main.yml` was not in the workflows folder. No need to create another PR
